app/routes/circulars/circulars.server.ts

```diff
@@ -0,0 +1,212 @@
+/*!
+ * Copyright © 2022 United States Government as represented by the Administrator
+ * of the National Aeronautics and Space Administration. No copyright is claimed
+ * in the United States under Title 17, U.S. Code. All Other Rights Reserved.
+ *
+ * SPDX-License-Identifier: NASA-1.3
+ */
+
+import { tables } from '@architect/functions'
+import type { DynamoDBDocument } from '@aws-sdk/lib-dynamodb'
+import { dynamoDBAutoIncrement } from '~/lib/dynamoDBAutoIncrement'
+import { getUser } from '../__auth/user.server'
+
+export type CircularModel = {
+  circularId?: number
+  sub?: string
+  createdOn: number
+  subject: string
+  body: string
+  submitter: string
+}
+
+export class CircularsServer {
+  #sub?: string
+  #email?: string
+  #groups?: string[]
+  #name?: string
+  #affiliation?: string
+  #isLocal: boolean
+
+  private constructor(
+    isLocal: boolean,
+    sub?: string,
+    email?: string,
+    groups?: string[],
+    name?: string,
+    affiliation?: string
+  ) {
+    this.#affiliation = affiliation
+    this.#email = email
+    this.#groups = groups
+    this.#name = name
+    this.#sub = sub
+    this.#isLocal = isLocal
+  }
+
+  static async create(request: Request) {
+    const user = await getUser(request)
+    const isLocal = new URL(request.url).hostname == 'localhost'
+    return new this(
+      isLocal,
+      user?.sub,
+      user?.email,
+      user?.groups,
+      user?.name,
+      user?.affiliation
+    )
+  }
+
+  /**
+   * Gets all available circulars
+   * @param searchString - a string that will be checked against subjects,
+   * bodies, and authors of saved circulars. A blank value returns all.
+   * @returns an array of CircularModel objects
+   */
+  async getCirculars(searchString?: string) {
+    let result = await this.#scanCircularsTable(searchString)
+    let items: CircularModel[] = result.Items
+    while (result.LastEvaluatedKey) {
+      result = await this.#scanCircularsTable(
+        searchString,
+        result.LastEvaluatedKey
+      )
+      items = [...items, ...result.Items]
+    }
+    return items.sort((a, b) => (b.circularId ?? 0) - (a.circularId ?? 0))
```

Comment:
I don't think that circularID can be undefined, can it?

```suggestion
    return items.sort((a, b) => b.circularId - a.circularId)
```